inc/plugins/namespace.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Sets DID-less plugin slug as active.
+ *
+ * @package FAIR
+ */
+
+namespace FAIR\Plugins;
+
+use function FAIR\Updater\get_packages;
+
+/**
+ * Bootstrap
+ *
+ * @return void
+ */
+function bootstrap() {
+	add_filter( 'option_active_plugins', __NAMESPACE__ . '\\set_as_active' );
+	add_filter( 'wp_admin_notice_markup', __NAMESPACE__ . '\\hide_notice', 10, 3 );
```

Comment:
Suggest only hooking this on `load-plugins.php` and such rather than hooking on every notice on every page.

Reply:
Are you certain that's where we want to limit it?

Reply:
Yeah that's the only file where the notice appears, where the deactivation error string is used, and the only place in Core where `validate_active_plugins()` or `validate_plugin()` are called.

Not being able to filter those functions is a real shame.